core/node/node_sync/src/batch_transaction_updater/mod.rs

```diff
@@ -0,0 +1,302 @@
+//! Component responsible for updating L1 batch status.
+
+use std::time::Duration;
+
+use tokio::sync::watch;
+use zksync_dal::{Connection, ConnectionPool, Core, CoreDal};
+use zksync_eth_client::EthInterface;
+use zksync_health_check::{Health, HealthStatus, HealthUpdater, ReactiveHealthCheck};
+use zksync_types::{
+    aggregated_operations::AggregatedActionType,
+    eth_sender::{EthTxFinalityStatus, L1BlockNumbers, TxHistory},
+    web3::TransactionReceipt,
+    Address, SLChainId,
+};
+
+use self::l1_transaction_verifier::L1TransactionVerifier;
+
+mod l1_transaction_verifier;
+
+#[cfg(test)]
+mod tests;
+
+/// Module responsible for updating L1 batch status. (Pending->FastFinalized->Finalized)
+/// Its currently the component that should terminate EN when gateway migration happens.
+/// The EN only is stopped when all batches from previous SL are finalized.
+#[derive(Debug)]
+pub struct BatchTransactionUpdater {
+    sl_client: Box<dyn EthInterface>,
+    l1_transaction_verifier: L1TransactionVerifier,
+    pool: ConnectionPool<Core>,
+    health_updater: HealthUpdater,
+    sleep_interval: Duration,
+    processing_batch_size: i64,
+}
+
+impl BatchTransactionUpdater {
+    pub fn new(
+        sl_client: Box<dyn EthInterface>,
+        diamond_proxy_addr: Address,
+        pool: ConnectionPool<Core>,
+        sleep_interval: Duration,
+        processing_batch_size: i64,
+    ) -> Self {
+        Self::from_parts(
+            sl_client,
+            diamond_proxy_addr,
+            pool,
+            sleep_interval,
+            processing_batch_size,
+        )
+    }
+
+    fn from_parts(
```

Comment:
Why is a separate method with the same signature as `new()` necessary?

Reply:
Leftover from a previous iteration when it was not. Fixing.